src/main/java/org/folio/rest/handlers/FeeFineBalanceChangedEventHandler.java

```diff
@@ -82,7 +81,34 @@ private void refreshOutstandingFeeFineBalance(UserSummary userSummary) {
   private Future<UserSummary> findSummaryByFeeFineIdOrFail(String feeFineId) {
     return userSummaryRepository.findByFeeFineId(feeFineId)
       .map(summary -> summary.orElseThrow(() -> new EntityNotFoundException(
-        format("User summary with feeFine %s was not found, event is ignored", feeFineId))));
+        format("User summary with fee/fine %s was not found, event is ignored", feeFineId))));
+  }
+
+  private void removeLoanIfLastLostItemFeeWasClosed(UserSummary userSummary,
+    FeeFineBalanceChangedEvent event) {
+
+    if (!isLostItemFeeId(event.getFeeFineTypeId())) {
+      return;
+    }
+
+    userSummary.getOpenLoans().stream()
+      .filter(OpenLoan::getItemLost)
+      .filter(loan -> StringUtils.equals(loan.getLoanId(), event.getLoanId()))
+      .findAny()
+      .ifPresent(loan -> {
+        boolean noLostItemFeesForLoanExist = userSummary.getOpenFeesFines().stream()
+          .filter(fee -> StringUtils.equals(fee.getLoanId(), event.getLoanId()))
+          .map(OpenFeeFine::getFeeFineTypeId)
+          .noneMatch(this::isLostItemFeeId);
+
+        if (noLostItemFeesForLoanExist) {
+          userSummary.getOpenLoans().remove(loan);
+        }
```

Comment:
I would extract this logic from the lambda to separate method

Reply:
This lambda doe not make much sense outside of this method, and the name of the variable IMHO makes it clear what the lambda does. So I'd keep it as is.